src/schema/resolvers/user.js

```diff
@@ -1,26 +1,59 @@
-const { v4: uuidv4 } = require('uuid');
+const { v4: uuidv4 } = require('uuid')
+const jwt = require('jsonwebtoken')
 
 export default {
-    Query: {
-        users: (parent, args, context, info) => context.models.user.findAll(),
-        user: (parent, args, context, info) => context.models.user.findByPk(args.id)
+  Query: {
+    users: (parent, args, context, info) => context.models.user.findAll(),
+    user: (parent, args, context, info) => context.models.user.findByPk(args.id),
+    currentUser: (parent, args, context, info) => context.user
+  },
+  Mutation: {
+    createUser: async (parent, args, context, info) => {
+      return createUserDB(args.input, context)
     },
-    Mutation: {
-        createUser: async (parent, args, context, info) => {
-            await console.log(context.models.user)
-            await console.log(args.input) // por alguna razon no tiene datos aunque le pase todo un inputUser
+    signup: async (parent, args, context, info) => {
 
-            var saltValue = context.models.user.getRandomSalt()
-            return context.models.user.create({
-                id: uuidv4(),
-                firstName: args.input.firstName,
-                lastName: args.input.lastName,
-                username: args.input.username,
-                salt : saltValue,
-                password : context.models.user.encryptPassword(args.input.password, saltValue)
-            })
-            //context.models.user.upsert
+      let newUser = await context.models.user.findOne({ where: { username: args.data.username } })
```

Comment:
There is a better way to write this check for an existing user first and then return a new one if it doesn't, let's continue in detail on Slack.